frame/crowdloan-rewards/src/lib.rs

```diff
@@ -480,38 +480,41 @@ where
 
 		if let Some(Call::associate { reward_account, proof }) = IsSubType::is_sub_type(call) {
 			if Associations::<T>::get(reward_account).is_some() {
-				return InvalidTransaction::Custom(ValidityError::AlreadyAssociated as u8).into();
+				return InvalidTransaction::Custom(ValidityError::AlreadyAssociated as u8).into()
 			}
 
 			let remote_account = match proof {
 				Proof::Ethereum(eth_proof) => {
 					let reward_account_encoded =
 						reward_account.using_encoded(|x| hex::encode(x).as_bytes().to_vec());
 					match ethereum_recover(T::Prefix::get(), &reward_account_encoded, eth_proof) {
-						Some(ethereum_address) => RemoteAccount::<<T as Config>::RelayChainAccountId>::Ethereum(ethereum_address),
-						None => {
+						Some(ethereum_address) =>
+							RemoteAccount::<<T as Config>::RelayChainAccountId>::Ethereum(
+								ethereum_address,
+							),
+						None =>
 							return InvalidTransaction::Custom(ValidityError::InvalidProof as u8)
-								.into()
-						}
+								.into(),
 					}
-				}
+				},
 				Proof::RelayChain(relay_account, relay_proof) => {
 					if verify_relay(
 						T::Prefix::get(),
 						reward_account,
 						relay_account.clone(),
 						relay_proof,
 					) {
-						RemoteAccount::<<T as Config>::RelayChainAccountId>::RelayChain(*reward_account)
+						RemoteAccount::<<T as Config>::RelayChainAccountId>::RelayChain(
+							relay_account.clone(),
+						)
 					} else {
-						return InvalidTransaction::Custom(ValidityError::InvalidProof as u8)
-							.into();
+						return InvalidTransaction::Custom(ValidityError::InvalidProof as u8).into()
 					}
-				}
+				},
 			};
 
 			if Rewards::<T>::get(remote_account).is_none() {
```

Comment:
I think it is cheaper to test this than checking for the proof first, we can perhaps move it before the proof checking?

Reply:
Do you mean just for the relay account?

Reply:
It's not a big deal, since you need to extract it from the signature, the code is currently nice and easy to understand, let's keep it like that :)